controllers/slicegateway/slicegateway.go

```diff
@@ -1534,3 +1547,60 @@ func (r *SliceGwReconciler) ReconcileIntermediateGatewayDeployments(ctx context.
 
 	return ctrl.Result{}, nil, false
 }
+
+// createPodDisruptionBudgetForSliceGatewayPods checks for PodDisruptionBudget objects in the cluster that match the
+// slice gateway pods, and if missing, it creates a PDB with minimum availability of 1 so at least one pod remains in
+// case of a disruption.
+func (r *SliceGwReconciler) createPodDisruptionBudgetForSliceGatewayPods(ctx context.Context, log logr.Logger,
+	sliceName string, sliceGateway *kubeslicev1beta1.SliceGateway) error {
+	log = log.WithValues("sliceName", sliceName, "sliceGwName", sliceGateway.Name)
+
+	// List PDBs in cluster that match the slice gateway pods
+	pdbs, err := listPodDisruptionBudgetForSliceGateway(ctx, r.Client, sliceName, sliceGateway.Name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, "failed to list PodDisruptionBudgets that match the slice gateway")
+
+		// When some unexpected error occurred, return the error for requeuing the request
+		return err
+	}
+
+	// Check if PDB already exists that matches the current slice gateway
+	if len(pdbs) > 0 {
+		log.Info("PodDisruptionBudget matching the slice gateway already exists. Skipping creation.")
+
+		return nil
+	}
+
+	// Create PDB manifest with minimum availability of 1 pod
+	pdb := constructPodDisruptionBudget(sliceName, sliceGateway.Name, DefaultMinAvailablePodsInPDB)
+
+	// Set SliceGateway instance as the owner and controller for PDB
+	if err = ctrl.SetControllerReference(sliceGateway, pdb, r.Scheme); err != nil {
+		log.Error(err, "Failed to set slice gateway as owner to PodDisruptionBudget",
+			"pdb", pdb.Name)
+
+		return fmt.Errorf("failed to set slice gateway %q as owner to PodDisruptionBudget %q: %v",
+			sliceGateway.Name, pdb.Name, err)
+	}
+
+	// Create PDB for slice gateway's pod to have at least 1 pod on each worker when disruption occurs
+	if err = r.Create(ctx, pdb); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			log.Info("PodDisruptionBudget matching the slice gateway already exists. Skipping creation",
```

Comment:
Plz remove this log. It will be noisy for Info level logs.

Reply:
Done. Thanks!